### PR TITLE
CI: clean azure pipeline

### PR DESCRIPTION
**Pull request by fangchenli**
We save about 2-3 min build time by using Azure's conda.

### CODE REVIEW

File: ci/azure/posix.yml [hunk 1]
```diff
@@ -73,24 +73,42 @@ jobs:
...
-        source activate pandas-dev
+    - bash: conda update --quiet --yes conda
+      displayName: Update conda
+
+    - bash: conda env create --quiet --file $(ENV_FILE)
+      displayName: Create Anaconda environment
+
+    - bash: |
+        eval "$(conda shell.bash hook)"
+        conda activate pandas-dev
```
Comment (fangchenli) [inline ci/azure/posix.yml:96]: Not sure why `source activate pandas-dev` doesn't work here.

Comment (jreback) [reply ci/azure/posix.yml:96]: source activate is the old way, conda activate is the new (both worked for a long time, but conda activate is the way)

File: ci/azure/posix.yml [hunk 2]
```diff
@@ -73,24 +73,42 @@ jobs:
...
+        sudo apt-get install -y libc6-dev-i386 $EXTRA_APT
       displayName: 'Install extra packages'
+      condition: eq('${{ parameters.name }}', 'Linux')
 
-    - script: echo '##vso[task.prependpath]$(HOME)/miniconda3/bin'
-      displayName: 'Set conda path'
+    - bash: echo '##vso[task.prependpath]$(HOME)/miniconda3/bin'
+      displayName: 'Add conda to PATH'
 
-    - script: ci/setup_env.sh
```
Comment (jreback) [inline ci/azure/posix.yml:85]: so this is basically inlining setup_env?

Comment (jreback) [reply ci/azure/posix.yml:85]: why are we doing this? that is called by every build

Comment (fangchenli) [reply ci/azure/posix.yml:85]: We save about 2-3 min build time by using the CI's conda. And the Windows builds are not using the script. We probably could combine all Azure CI setups into one file. Now we have one for posix and one for windows. They are almost identical.

Comment (jreback) [reply ci/azure/posix.yml:85]: ok the problem is there are *other* things that setup_env is doing which this is not. not averse to this, but we need to do it *inside* setup_env.sh

File: azure-pipelines.yml [hunk 3]
```diff
@@ -11,18 +11,17 @@ variables:
   PYTEST_WORKERS: auto
 
 jobs:
-# Mac and Linux use the same template
-- template: ci/azure/posix.yml
+- template: ci/azure/macos.yml
   parameters:
     name: macOS
-    vmImage: macOS-10.14
+    vmImage: macOS-10.15
```
Comment (jbrockmendel) [inline azure-pipelines.yml:17]: @fangchenli can any of this be split off into an easy preliminary?

Comment (fangchenli) [reply azure-pipelines.yml:17]: #42419

**Pull request closed without merging**